data/yam/agama/auto/lib/storage.libsonnet

```diff
@@ -0,0 +1,45 @@
+local get_root_filesystem(filesystem) = {
+  drives: [
+    {
+      partitions: [
+        { search: "*", delete: true },
+        { generate: 'default' },
+        { filesystem: { path: '/', type: filesystem } },
+      ],
+    },
+  ],
+};
+
+{
+  lvm(encrypted=false): {
+    drives: [
+      {
+        alias: 'pvs-disk',
+        partitions: [
+          { search: "*", delete: true }
+        ]
+      },
+    ],
+    volumeGroups: [
+      {
+        name: 'system',
+        physicalVolumes: [
+          {
+            [if encrypted == true then 'generate']: {
+              targetDevices: ['pvs-disk'],
+              encryption: {
+                luks2: { password: "nots3cr3t" }
+              }
+            },
+            [if encrypted == false then 'generate']: ['pvs-disk'],
+          },
+        ],
+        logicalVolumes: [
+          { generate: 'default' },
+        ],
+      },
+    ]
+  },
+  root_filesystem_ext4: get_root_filesystem('ext4'),
```

Comment:
these getter/setters from OOP confuse me a bit here, I would keep it simple, because there are not behaviors it is just configuration, so `root_filesystem` would do the job.